services/src/main/java/org/keycloak/authorization/authorization/AuthorizationTokenService.java

```diff
@@ -517,7 +516,7 @@ private void resolvePreviousGrantedPermissions(PermissionTicketToken ticket,
                             break;
                         }
 
-                        Resource resource = resourceStore.findById(grantedPermission.getResourceId(), ticket.getIssuedFor());
```

Comment:
I think the `issuedFor` is the client id, isn't it? I can't remember now if we translate it to the client's corresponding ID or if we actually lookup using the client id.

Reply:
`issuedFor` is `client.clientId` not `client.id` [1]. It worked because `findById` methods in cache and jpa don't check `resourceServerId` when querying. However, the new storage does so no Resource is found in storage. The method works only if the Resource with id is present in the cache.

[1] https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/authorization/admin/PolicyEvaluationService.java#L293